### DISCUSSION TITLE
'thresh' parameter when using DiehlAndCook2015 model?

### DISCUSSION THREAD

**Discussion by rafaelblevin821** [BindsNET/bindsnet#645] [General]
Hi, I am trying to change the `thresh` value for `DiehlAndCook2015` model, but it does not seem to be a callable argument.

I added it to the argument parsers in the 'models' file, hoping this would solve the problem, but I still get the following message:

`usage: sprvsd_cancer_weights_sum_sam_00.py [-h] [--seed SEED] [--n_neurons N_NEURONS] [--n_epochs N_EPOCHS]
                                           [--n_train N_TRAIN] [--n_test N_TEST] [--n_clamp N_CLAMP] [--exc EXC]
                                           [--inh INH] [--theta_plus THETA_PLUS] [--time TIME] [--dt DT]
                                           [--intensity INTENSITY] [--progress_interval PROGRESS_INTERVAL]
                                           [--update_interval UPDATE_INTERVAL] [--train] [--test] [--plot] [--gpu]
                                           [--device_id DEVICE_ID]
sprvsd_cancer_weights_sum_sam_00.py: error: unrecognized arguments: --thresh 50`

I am sure I must be missing something...

Any help on this would be greatly appreciated.

**Comment by Hananel-Hazan** [comment]
I am not familiar with the script `sprvsd_cancer_weights_sum_sam_00.py`, The error message seems to come from that script. 

In any case, the last updated the `DiehlAndCook2015` to added two more parameters `inh_thresh` default `40.0` and `exc_thresh` default `-52.0` to accommodate the two neuron population (inhibitory and excitatory).

**Reply by rafaelblevin821** [reply]
@Hananel-Hazan thanks.  

`inh_thresh` and `exc_thresh` aren't in the bindsnet version 0.3.1 I am using. Is there any way to update without causing problems? 

I can add the extra lines into the `DiehlAndCook2015` model in `models.py`, but I'm not sure if any other files also need to be changed...?

**Reply by Hananel-Hazan** [reply]
Depend how you are using BindsNET. If you installed using the pip install command you only need to run again `pip install -U git+https://github.com/BindsNET/bindsnet.git` . If you use the development installation, just update the repository and you all set.

**Reply by rafaelblevin821** [reply]
I think I have it working by adding the `inh_thresh` and `exc_thresh` to models.py and adding the argument parsers. Any further issues, I will update.

Thanks @Hananel-Hazan
